topics/icrc_28_trusted_origins.md

```diff
@@ -37,38 +39,46 @@ relying party from another.
 > number if a port is present and differs from the default port for the given protocol. Examples of typical origins
 > are https://example.org (implying port 443), http://example.net (implying port 80), and http://example.com:8080.
 
-The origin can't be trusted with some transport layers. It's up to the signer to implement a trustable way to identify
-the origin of the relying party.
+When the signer obtains the origin of the relying party, it **MUST** make sure that it's genuine. Which means that the
+transport channel used between the relying party and signer, must return a origin that can be trusted to be genuine
+e.g. [ICRC-29](./icrc_29_window_post_message_transport.md).
 
-The origin **MUST** be identified in a trustable way by the signer for example through `postMessage`.
+## Verify
 
-## Get trusted origins
+When the signer receives a delegation chain request for a global identity, it must verify that the canister targets
+trust the relying party making the request. The signer can use the below method to get a list of trusted origins for
+each canister target and then verify if the relying party is within each list.
 
-A canister **MUST** make both a metadata entry and a canister method available, both **MUST** return the same list of
-trusted origins.
+### icrc28_get_trusted_origins
 
-### Canister metadata
+Returns a list of origins trusted by the canister.
 
-Allows for tooling to pull the list of trusted origins out of the state tree from metadata
-section `icp:public icrc28_trusted_origins`.
+```
+icrc28_get_trusted_origins : () -> (vec text);
```

Comment:
If we change the return type to `(record {trusted_origins: vec text})` we'll have a much easier time extending it later. And we also get a name for the values. 😉